### PR TITLE
Test raw autocomplete api examples

### PR DESCRIPTION
**Pull request by derdilla**
Part of #130459.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: examples/api/test/widgets/autocomplete/raw_autocomplete.focus_node.0_test.dart [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      matching: find.byType(TextFormField)
+    ), findsOneWidget);
+
+    expect(find.byType(RawAutocomplete<String>), findsOneWidget);
+    expect(find.text('aardvark'), findsNothing);
+    expect(find.text('bobcat'), findsNothing);
+    expect(find.text('chameleon'), findsNothing);
+    expect(find.ancestor(
+        matching: find.byType(AppBar),
+        of: find.byType(RawAutocomplete)
```
Comment (bleroux) [inline examples/api/test/widgets/autocomplete/raw_autocomplete.focus_node.0_test.dart:26]: Too much indentation

File: examples/api/test/widgets/autocomplete/raw_autocomplete.1_test.dart [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    await tester.pump();
+
+    expect(find.byType(ListTile), findsNothing);
+    expect(find.descendant(
+      of: find.byType(TextFormField),
+      matching: find.text('Bob'),
+    ), findsOneWidget);
+  });
+
+  testWidgets('Finds users by mail address', (WidgetTester tester) async {
```
Comment (bleroux) [inline examples/api/test/widgets/autocomplete/raw_autocomplete.1_test.dart:48]: mail -> email

**Review by bleroux — APPROVED** [review]
LGTM with minor nits.
Thanks!

File: examples/api/test/widgets/autocomplete/raw_autocomplete.focus_node.0_test.dart [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+void main() {
+  testWidgets('Autocomplete example is visible', (WidgetTester tester) async {
+    await tester.pumpWidget(const example.AutocompleteExampleApp());
+    expect(find.text('Split RawAutocomplete App'), findsOneWidget);
+    expect(find.byType(AppBar), findsOneWidget);
+    expect(find.byType(TextFormField), findsOneWidget);
+    expect(find.descendant(
+      of: find.byType(AppBar),
+      matching: find.byType(TextFormField)
+    ), findsOneWidget);
```
Comment (TahaTesser) [inline examples/api/test/widgets/autocomplete/raw_autocomplete.focus_node.0_test.dart:18]:
```suggestion
    expect(
      find.descendant(
        of: find.byType(AppBar),
        matching: find.byType(TextFormField)
      ),
      findsOneWidget,
    );
```

File: examples/api/test/widgets/autocomplete/raw_autocomplete.focus_node.0_test.dart [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    expect(find.byType(RawAutocomplete<String>), findsOneWidget);
+    expect(find.text('aardvark'), findsNothing);
+    expect(find.text('bobcat'), findsNothing);
+    expect(find.text('chameleon'), findsNothing);
+    expect(find.ancestor(
+      matching: find.byType(AppBar),
+      of: find.byType(RawAutocomplete)
+    ), findsNothing);
```
Comment (TahaTesser) [inline examples/api/test/widgets/autocomplete/raw_autocomplete.focus_node.0_test.dart:27]:
```suggestion
    expect(
      find.ancestor(
        matching: find.byType(AppBar),
        of: find.byType(RawAutocomplete)
      ),
      findsNothing,
    );
```

File: examples/api/test/widgets/autocomplete/raw_autocomplete.focus_node.0_test.dart [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    expect(find.text('chameleon'), findsNothing);
+
+    await tester.tap(find.text('bobcat'));
+    await tester.pump();
+
+    expect(find.byType(ListTile), findsNothing);
+    expect(find.descendant(
+      of: find.byType(TextFormField),
+      matching: find.text('bobcat'),
+    ), findsOneWidget);
```
Comment (TahaTesser) [inline examples/api/test/widgets/autocomplete/raw_autocomplete.focus_node.0_test.dart:55]:
```suggestion
    expect(
      find.descendant(
        of: find.byType(TextFormField),
        matching: find.text('bobcat'),
      ),
      findsOneWidget,
    );
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]
Thanks for the tests!

**Review by TahaTesser — APPROVED** [review]

**Comment by TahaTesser** [comment]
@NobodyForNothing 
You might need to rebase to fix Google testing

**Comment by bleroux** [comment]
@NobodyForNothing Can you rebase this PR?

**Pull request merged**